src/common/signer.ts

```diff
@@ -125,6 +130,26 @@ export abstract class TurboDataItemAbstractSigner
     amount,
     gatewayUrl,
   }: SendTxWithSignerParams): Promise<string> {
+    if (this.walletAdapter) {
+      if (!isEthereumWalletAdapter(this.walletAdapter)) {
+        throw new Error(
+          'Unsupported wallet adapter -- must implement getSigner',
+        );
+      }
+      const signer = this.walletAdapter.getSigner();
+      if (signer.sendTransaction === undefined) {
+        throw new Error(
+          'Unsupported wallet adapter -- getSigner must return a signer with sendTransaction API for crypto funds transfer',
+        );
+      }
+
+      const { hash } = await signer.sendTransaction({
+        to: target,
+        value: parseEther(amount.toFixed(18)),
+      });
+      return hash;
+    }
```

Comment:
here we defer to ethereum wallet adapter when available which allows web wallets to use the crypto-fund flow